README.md

```diff
@@ -3,132 +3,339 @@
+2. create a build folder `build`, run CMake, and run `ccmake` for editiong the CMake variables
+```
+mkdir buid
+cd build
+cmake ..
+ccmake .
+```
+3. Set the `CMAKE_INSTALL_PREFIX` to your `$ROBOTOLOGY_SUPERBUILD_INSTALL_PREFIX` if you're using the superbuild, or any other location at your convenience.
+4. Set the `ICUB_MODELS_TO_INSTALL`, which is by default set to `iCubGazeboV2_5`.
+5. Turn `INSTALL_FAKE_BATTERY_DEVICE_CONFIG_FILES` on. Actually, the CMake configuration in the repository is for installing any device of additional module to build through the CMake system. In such case, installing the fake battery device can still be optional.
+6. configure and generate the CMake files (hit "c" as many times as required, then "g").
+7. install
+```
+make install
+``` 
+
 ### Notes
-- For checking the active node.js/npm versions, run respectively `node -v` and `npm -v`.
-- The installation of Open MCT dependency completes with a warning on (refer to https://github.com/dic-iit/element_software-engineering/issues/47#issuecomment-855276088).
+- For checking the active Node.js/npm versions, run respectively `node -v` and `npm -v`.
+- The installation of Open MCT dependency completes with a warning on some detected network vulnerability (refer to https://github.com/ami-iit/yarp-openmct/issues/35). This is not critical as we are running everything in a local private network, but further analysis is required.
 
 
 ## [Client Installation](#top)
 
 Install one of the browsers listed in the client dependencies.
 
 
-## [How to Run the Server](#top)
+## [How to Run the Telemetry Server](#top)
+
+### Run a YARP Robot interface or Connect to a Running Robot Network
+
+For running a simulation on the machine where you installed the simulator and dependencies:
+
+1. Activate the `robotologyenv` environment
+   ```
+   conda activate robotologyenv
+   ```
+2. Set the YARP_ROBOT_NAME environment variable according to the chosen Gazebo model (`icubGazeboSim` or `iCubGazeboV2_5`):
+   ```
+   export YARP_ROBOT_NAME="iCubGazeboV2_5"
+   ```
+2. Set the YARP namespace and run the YARP server
+   ```
+   yarp namespace /myNameSpace
+   yarpserver --write
+   ```
+3. Run Gazebo and drag and drop the iCub model (`icubGazeboSim` or `iCubGazeboV2_5`):
+   ```
+   gazebo -slibgazebo_yarp_clock.so
+   ```
+
+If you wish to visualize iCub telemetry data from a robot setup already running on its local network under a given namespace \<yarpRobotNamespace\> (e.g. `/icub`, `/myNameSpace`), follow the steps below:
+
+1. Set the YARP_ROBOT_NAME environment variable according to the target model \<robotModel\> (real robot model `iCubGenova04`, `iCubGenova09` model or simulation model `icubGazeboSim` or `iCubGazeboV2_5`):
+   ```
+   export YARP_ROBOT_NAME="<robotModel>"
+   ```
+2. Set the Yarp namespace, detect the Yarp name server and save its address:
+   ```
+   yarp namespace <yarpRobotNamespace>
+   yarp detect --write
+   ```
```

Comment:
I was expecting this part to be generic (how to run the server), but actually applies only to the simulation case, which is only for testing.

Reply:
Refactored.